cypress/integration/unit_tests/checkbox_test.js

```diff
@@ -92,35 +101,238 @@ describe('Unit test for toggleLayerOn', () => {
   });
 
   it('check hidden layer, then uncheck before list evaluation', () => {
-    expect(layerArray[2].displayed).to.equals(false);
+    expect(layerArray[2].displayed).to.be.false;
     expect(layerArray[2].data).to.be.undefined;
 
     const emptyList = [];
     let callback = null;
     stubForEmptyList((callb) => callback = callb);
-    toggleLayerOn(mockFirebaseLayers[2]);
+    const promise = toggleLayerOn(mockFirebaseLayers[2]);
+    expect(promise).to.not.be.null;
+    expect(layerArray[2].displayed).to.be.true;
+    expect(layerArray[2].data).to.be.undefined;
     toggleLayerOff(2);
     callback(emptyList);
-
-    // Evaluate after the promise finishes by using an instant wait.
-    cy.wait(0).then(() => {
-      expect(layerArray[2].displayed).to.equals(false);
+    cy.wrap(promise).then(() => {
+      expect(layerArray[2].displayed).to.be.false;
       expect(layerArray[2].data).to.not.be.null;
       const layerProps = deckGlArray[2].props;
       expect(layerProps).to.have.property('id', 'asset2');
       expect(layerProps).to.have.property('visible', false);
       expect(layerProps).to.have.property('data', emptyList);
     });
   });
+
+  it('checks hidden layer, unchecks, checks before list evaluation', () => {
```

Comment:
can you specify what list you're talking about here/why it's relevant

Reply:
Reworded. The test itself does setup for the list evaluation, since EE features are evaluated as a list.